src/TestFramework/TestFramework/Attributes/TestMethod/RetryContext.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+public sealed class RetryContext
```

Comment:
Shall we use `struct` like roslyn? We probably have very few callback so it might not be needed.

Reply:
Fine for me to use struct.